### PR TITLE
docs: Update desktop publishing article

### PR DESCRIPTION
**Pull request by ebariche**
## PR Type

What kind of change does this PR introduce?
- Documentation

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

### CODE REVIEW

File: doc/articles/uno-publishing-desktop.md [hunk 1]
```diff
@@ -2,50 +2,120 @@
 uid: uno.publishing.desktop
 ---
 
-# Publishing Your App for Desktop
+# Publishing Your App For Desktop
 
 ## Preparing For Publish
 
-- [Profile your app with VS 2022](https://learn.microsoft.com/en-us/visualstudio/profiling/profiling-feature-tour?view=vs-2022)
+- [Profile your app with Visual Studio](https://learn.microsoft.com/en-us/visualstudio/profiling)
```
Comment (morning4coffe-dev) [inline doc/articles/uno-publishing-desktop.md:9]:
```suggestion
- [Profile your app with Visual Studio](https://learn.microsoft.com/visualstudio/profiling)
```

File: doc/articles/uno-publishing-desktop.md [hunk 2]
```diff
@@ -2,50 +2,117 @@
...
+- core24 on Ubuntu 24.04
+- multipass
+- lxd
+  - current user must be part of the `lxd` group
+  - `lxd init --minimal` or similar should be run
+- snapcraft
+
+Docker may interfere with Lxd causing network connectivity issues, for solutions see:
+
+https://documentation.ubuntu.com/lxd/en/stable-5.0/howto/network_bridge_firewalld/#prevent-connectivity-issues-with-lxd-and-docker
```
Comment (jeromelaban) [inline doc/articles/uno-publishing-desktop.md:85]:
```suggestion
> [!NOTE]
> Docker may interfere with Lxd causing network connectivity issues, [read this article for a solution](https://documentation.ubuntu.com/lxd/en/stable-5.0/howto/network_bridge_firewalld/#prevent-connectivity-issues-with-lxd-and-docker).
```

File: doc/articles/uno-publishing-desktop.md [hunk 3]
```diff
@@ -2,50 +2,117 @@
...
+We support creating .snap packages on **Ubuntu 20.04** or later.
+
+#### Requirements
+
+The following must be installed and configured:
+
+- snapd
+- core20 on Ubuntu 20.04
+- core22 on Ubuntu 22.04
+- core24 on Ubuntu 24.04
```
Comment (jeromelaban) [inline doc/articles/uno-publishing-desktop.md:76]: Let's explain how to do this, or point to docs that do so.

File: doc/articles/uno-publishing-desktop.md [hunk 4]
```diff
@@ -2,50 +2,117 @@
...
 
-- [ClickOnce](https://learn.microsoft.com/visualstudio/deployment/quickstart-deploy-using-clickonce-folder?view=vs-2022) on Windows
-- Using a Zip file, then running the app using `dotnet [yourapp].dll`
+To publish a snap, run:
+
+```shell
+dotnet publish -f net8.0-desktop -r {{RID}} -p:SelfContained=true -p:PackageFormat=snap
+```
+
+Where {{RID}} is either `linux-x64` or `linux-arm64`.
```
Comment (jeromelaban) [inline doc/articles/uno-publishing-desktop.md:95]:
```suggestion
Where `{{RID}}` is either `linux-x64` or `linux-arm64`.
```

File: doc/articles/uno-publishing-desktop.md [hunk 5]
```diff
@@ -2,50 +2,117 @@
...
+
+We generate snap manifests in classic confinement mode and a .desktop file by default.
+
+If you wish to customize your snap manifest, you will need to pass the following MSBuild properties:
+
+- SnapManifest
+- DesktopFile
```
Comment (jeromelaban) [inline doc/articles/uno-publishing-desktop.md:102]: Should we point to docs on both file formats?

File: doc/articles/uno-publishing-desktop.md [hunk 6]
```diff
@@ -2,50 +2,117 @@
...
+
+The .desktop filename MUST conform to the Desktop File spec.
```
Comment (jeromelaban) [inline doc/articles/uno-publishing-desktop.md:104]:
```suggestion
The `.desktop` filename MUST conform to the Desktop File spec.
```

File: doc/articles/uno-publishing-desktop.md [hunk 7]
```diff
@@ -2,50 +2,117 @@
...
+
+If you wish, you can generate a default snap manifest and desktop file, then tweak them.
```
Comment (jeromelaban) [inline doc/articles/uno-publishing-desktop.md:106]: How ?

File: doc/articles/uno-publishing-desktop.md [hunk 8]
```diff
@@ -2,50 +2,117 @@
...
+
+> [!NOTE]
+> .NET 9 publishing and cross-publishing are not supported yet. We will support .NET 9 publishing soon.
```
Comment (jeromelaban) [inline doc/articles/uno-publishing-desktop.md:109]:
```suggestion
> .NET 9 publishing and cross-publishing are not supported as of Uno 5.5.
```

**Comment by jeromelaban** [comment]
@mergifyio backport release/stable/5.5

**Review by jeromelaban — APPROVED** [review]

**Pull request merged**